src/Microsoft.Extensions.Logging.EventSource/EventSourceLoggerFactoryExtensions.cs

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Extensions.Logging.EventSourceLogger;
+
+namespace Microsoft.Extensions.Logging
+{
+    /// <summary>
+    /// Extension methods for the <see cref="ILoggerFactory"/> class.
+    /// </summary>
+    public static class EventSourceLoggerFactoryExtensions
+    {
+        /// <summary>
+        /// Adds an event logger that is enabled for <see cref="LogLevel"/>.Information or higher.
+        /// </summary>
+        /// <param name="factory">The extension method argument.</param>
+        public static ILoggerFactory AddEventSourceLogger(this ILoggerFactory factory)
+        {
+            if (factory == null)
+            {
+                throw new ArgumentNullException(nameof(factory));
+            }
+
+            LoggingEventSource.Instance.CreateLoggerProvider(factory);
+
+            // Do not call factory.AddProvider() here. EventSourceLoggerProvider will do that automatically 
+            // when something starts using Microsoft-Extensions-Logging EventSource. This way there is no overhead
+            // from using the EventSourceLogger when no one is interested in the data it exposes.
```

Comment:
LoggingEventSource is perhaps different from most loggers in that the creation of a LoggingEvnetSource indicates the POTENTIAL DESIRE for logging (you want to allow users to get at your events via EventListeners / ETW).   It is only later, when someone actually turns on logging using EventListener/ETW that you actually log anything.   It would be a shame to have to call into the LoggingEventSource on every event just to drop the data.   

Logically what we are going is dynamically adding a LoggingProvider when an EventListener tells us to.  That seems like a perfectly valid thing to do, and it avoids ANY overhead when it the EventSource is off.